ui/pages/swaps/view-quote/view-quote.js

```diff
@@ -755,14 +794,18 @@ export default function ViewQuote() {
         baseAndPriorityFeePerGas === undefined) ||
       (!networkAndAccountSupports1559 &&
         (gasPrice === null || gasPrice === undefined)) ||
-      (currentSmartTransactionsEnabled && currentSmartTransactionsError),
+      (currentSmartTransactionsEnabled &&
+        (currentSmartTransactionsError || smartTransactionsError)) ||
+      (currentSmartTransactionsEnabled &&
+        smartTransactionsOptInStatus &&
+        !smartTransactionFees?.tradeTxFees),
   );
 
   useEffect(() => {
     if (
       currentSmartTransactionsEnabled &&
       smartTransactionsOptInStatus &&
```

Comment:
I know this isn't newly added but can you explain the difference between `currentSmartTransactionsEnabled` and `smartTransactionsOptInStatus`?

Reply:
Sure! So, `currentSmartTransactionsEnabled` tells us if STX are enabled in UI. That depends on multiple things, like is it Ethereum mainnet, no HW wallet, feature flag for STX on, etc.

`smartTransactionsOptInStatus` indicates if a user opted into STX or if they want to use regular swaps.

Reply:
Maybe I'm misunderstanding but it sounds like `smartTransactionsOptInStatus ` is contained within `currentSmartTransactionsEnabled`?  Is it not redundant somehow?

Reply:
`currentSmartTransactionsEnabled` doesn't contain the Opt In check. If it's true, you can see the STX switch in UI: 
![image](https://user-images.githubusercontent.com/80175477/183704075-ca59db33-9107-43e0-bc61-75232644ab20.png)

But if you turn off Smart transaction, `smartTransactionsOptInStatus` will be false:
![image](https://user-images.githubusercontent.com/80175477/183704161-c02226af-c620-4a2c-accc-7fed590e991e.png)

Reply:
OH sorry I see I was confusing enabled via the feature flag with enabled by this toggle in the UI.